plugin/storage/memory/memory_test.go

```diff
@@ -523,3 +523,260 @@ func makeTestingSpan(traceID model.TraceID, suffix string) *model.Span {
+						ServiceName: "serverService",
+					},
+					References: []model.SpanRef{
+						{
+							RefType: model.ChildOf,
+							TraceID: clientSpan.TraceID,
+							SpanID:  clientSpan.SpanID,
+						},
+					},
+				}
+				return []*model.Span{clientSpan, serverSpan}
+			},
+			endTs:    time.Now().Add(1 * time.Second),
+			lookback: time.Hour,
+			expectedDeps: []model.DependencyLink{
+				{Parent: "clientService", Child: "serverService", CallCount: 1},
+			},
+		},
+	}
+
+	for _, scenario := range scenarios {
+		t.Run(scenario.description, func(t *testing.T) {
+			store := NewStore()
+			spans := scenario.setupSpans()
+
+			for _, span := range spans {
+				assert.NoError(t, store.WriteSpan(context.Background(), span))
+			}
+
+			dependencies, err := store.GetDependencies(context.Background(), scenario.endTs, scenario.lookback)
+			assert.NoError(t, err)
+			assert.Equal(t, scenario.expectedDeps, dependencies, "Dependencies do not match the expected output")
+		})
+	}
+}
+
+func TestIsLeaf(t *testing.T) {
+	scenarios := []struct {
+		description string
+		spans       []*model.Span
+		targetSpan  *model.Span
+		expected    bool
+	}{
+		{
+			description: "Span with no children should be a leaf",
+			spans: []*model.Span{
+				{SpanID: model.NewSpanID(1)},
+			},
+			targetSpan: &model.Span{SpanID: model.NewSpanID(1)},
+			expected:   true,
+		},
+		{
+			description: "Span with a child should not be a leaf",
+			spans: []*model.Span{
+				{
+					SpanID: model.NewSpanID(1),
+				},
+				{
+					SpanID: model.NewSpanID(2),
+					References: []model.SpanRef{
+						{
+							RefType: model.ChildOf,
+							TraceID: model.NewTraceID(1, 2),
+							SpanID:  model.NewSpanID(1),
+						},
+					},
+				},
+			},
+			targetSpan: &model.Span{SpanID: model.NewSpanID(1)},
+			expected:   false,
+		},
+		{
+			description: "Span with no direct children in a list should be a leaf",
+			spans: []*model.Span{
+				{SpanID: model.NewSpanID(1)},
+				{SpanID: model.NewSpanID(2)},
+			},
+			targetSpan: &model.Span{SpanID: model.NewSpanID(1)},
+			expected:   true,
+		},
+	}
+
+	for _, scenario := range scenarios {
+		t.Run(scenario.description, func(t *testing.T) {
+			result := isLeaf(scenario.targetSpan, scenario.spans)
+			assert.Equal(t, scenario.expected, result, "Unexpected result for isLeaf in scenario '%s'", scenario.description)
+		})
+	}
+}
+
+func TestInferServiceName(t *testing.T) {
+	scenarios := []struct {
+		tags         model.KeyValues
+		isLeaf       bool
+		expectedName string
+	}{
+		{
+			tags: model.KeyValues{
+				model.String("peer.service", "authService"),
+			},
+			expectedName: "authService",
+		},
+		{
+			tags: model.KeyValues{
+				model.String("rpc.service", "grpcService"),
+			},
+			expectedName: "rpc-grpcService",
+		},
+		{
+			tags: model.KeyValues{
+				model.String("http.route", "users"),
+			},
+			expectedName: "http-users",
+		},
+		{
+			tags: model.KeyValues{
+				model.String("db.system", "mysql"),
```

Comment:
add a test for blank to increase code coverage

Reply:
Done!